pkg/nodediscovery/nodediscovery.go

```diff
@@ -87,13 +88,13 @@ func enableLocalNodeRoute() bool {
 }
 
 // NewNodeDiscovery returns a pointer to new node discovery object
-func NewNodeDiscovery(manager *nodemanager.Manager, mtuConfig mtu.Configuration, netConf *cnitypes.NetConf) *NodeDiscovery {
+func NewNodeDiscovery(manager *nodemanager.Manager, mtuConfig mtu.Configuration, netConf *cnitypes.NetConf) (*NodeDiscovery, error) {
 	auxPrefixes := []*cidr.CIDR{}
 
 	if option.Config.IPv4ServiceRange != AutoCIDR {
 		serviceCIDR, err := cidr.ParseCIDR(option.Config.IPv4ServiceRange)
 		if err != nil {
-			log.WithError(err).WithField(logfields.V4Prefix, option.Config.IPv4ServiceRange).Fatal("Invalid IPv4 service prefix")
+			return nil, fmt.Errorf("Invalid IPv4 service prefix", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("invalid IPv4 service prefix: %v", err)
```

Reply:
Did you mean `%w`? I think we should generally be wrapping errors properly rather than formatting them when passing them up anywhere in the codebase?

Reply:
Ah yes I did mean that.

Reply:
hold on, this file wasn't supposed to make it into the edited set. Let me back out these changes, nodediscovery.go is another topic, since it fatals in a go routine, which I need to discuss in a separate PR :|